File: LC/lc206/lc206.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
  var res = new ListNode();
  while(head){
      let tmp = res.next;
      res.next = new ListNode(head.val);
      res.next.next = tmp;
      head = head.next;
  }
  return res.next;
};

File: LC/lc206/lc206.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        def rev_aux (head, res):
            if head:
                res = ListNode(head.val, res)
                return rev_aux(head.next, res)
            return res
        return rev_aux(head, None)
        